cpp/src/io/orc/writer_impl.cu

```diff
@@ -1201,41 +1163,67 @@ encoded_data encode_columns(orc_table_view const& orc_table,
   }
   dictionaries.data.clear();
   dictionaries.index.clear();
-  stream.synchronize();
+  chunk_streams.device_to_host(stream, true);
```

Comment:
Nit: Whenever passing literal value, please always try to add comment saying what it is. Otherwise others may not know what is it.
```suggestion
  chunk_streams.device_to_host(stream, true /* device sync */);
```